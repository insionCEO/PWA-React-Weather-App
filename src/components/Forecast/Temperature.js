
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ForeCastTemperature extends Component {
  render() {
    return (
      <p className="forecast__temperature">{this.props.temperature}
        <span className="temperature__unit-dot"></span>
      </p>
    )
  }
}

ForeCastTemperature.propTypes = {
  temperature: PropTypes.string.isRequired
};

export default ForeCastTemperature;
