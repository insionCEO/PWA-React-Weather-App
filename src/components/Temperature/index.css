.current-condition {
  text-align: center;
}

.temperature {
  align-items: center;
  display: flex;
  justify-content: center;
}

.temperature__value {
  font-size: 196px;
}

.temperature__unit {
  display: flex;
  flex-direction: row;
}

.temperature__status {
  font-size: 24px;
  position: absolute;
  text-align: center;
  width: 100%;
}

.temperature__unit .temperature__unit-dot {
  border-radius: 6px;
  height: 6px;
  margin-left: -9px;
  margin-top: -59px;
  position: absolute;
  width: 6px;
}

.temperature__unit-dot {
  border: 2px solid #ffffff;
  border-radius: 10px;
  height: 10px;
  width: 10px;
}

.temperature__unit-letter {
  font-size: 35px;
  font-weight: 100;
  margin-left: 7px;
  margin-top: -68px;
  position: absolute;
}
