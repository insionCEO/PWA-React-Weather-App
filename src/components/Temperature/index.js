import React, { Component } from 'react';
import rAFTimeout from '../../helpers/rAFTimeout';
import './index.scss';

class Temperature extends Component {
  animate() {
    rAFTimeout(() => this.status.classList.add('animate-in'), 150);
    rAFTimeout(() => this.value.classList.add('animate-in'), 250);
    rAFTimeout(() => this.unit.classList.add('animate-in'), 300);
  }

  componentDidMount() {
    this.status = document.querySelector('.temperature__status');
    this.value = document.querySelector('.temperature__value');
    this.unit = document.querySelector('.temperature__unit');

    rAFTimeout(() => this.animate(), 400);
  }

  render() {
    return (
      <section className="current-condition">
        <div className="wrapper-temperature">
          <span className="temperature__status">{this.props.weather}</span>
          <div className="temperature">
            <span className="temperature__value">{this.props.temperature}</span>
            <div className="temperature__unit">
              <span className="temperature__unit-dot"></span>
              <span className="temperature__unit-letter">c</span>
            </div>
          </div>
        </div>
      </section>
    )
  }
}

export default Temperature;
