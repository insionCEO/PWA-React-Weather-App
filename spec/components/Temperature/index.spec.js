import React from 'react';
import Temperature from '../../../src/components/Temperature';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('Temperature', () => {
  it('matches snapshot', () => {
    const component = mount(<Temperature />);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls animate', (done) => {
      const component = mount(<Temperature />);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 500);
    });
  });
});
