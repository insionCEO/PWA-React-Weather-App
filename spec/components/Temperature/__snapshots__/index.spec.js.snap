// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature matches snapshot 1`] = `
<Temperature>
  <section
    className="current-condition"
  >
    <div
      className="wrapper-temperature"
    >
      <span
        className="temperature__status"
      />
      <div
        className="temperature"
      >
        <span
          className="temperature__value"
        />
        <div
          className="temperature__unit"
        >
          <span
            className="temperature__unit-dot"
          />
          <span
            className="temperature__unit-letter"
          >
            c
          </span>
        </div>
      </div>
    </div>
  </section>
</Temperature>
`;
